src/auth0/handlers/roles.js

```diff
@@ -121,4 +128,23 @@ export default class RoleHandler extends DefaultHandler {
       }
     }));
   }
+
+  async checkTenantSupportsRoles() {
+    try {
+      await this.client.roles.get({ id: 'rol_0000000000000000' });
+      return true;
+    } catch (error) {
+      // this means the tenant does not have roles implemented yet
+      if (error.statusCode === 501) {
+        log.info(`Role processing ignored - ${error.message}`);
+        return false;
+      }
+      // tenant api calls for get role works but role does not exist or some other error. This is ok!
+      if ([ 404, 401, 403, 429 ].indexOf(error.statusCode) >= 0) {
```

Comment:
should this just be all `400` level status codes? why was this explicit list picked?

if we keep the list, can they be properly ordered?